cmd/demo/payment.go

```diff
@@ -106,6 +109,7 @@ func (ch *paymentChannel) Handle(update client.ChannelUpdate, res *client.Update
 		ch.log.Trace("Calling onFinal handler for paymentChannel")
 		go ch.onFinal()
 	}
+	ch.lastState = update.State.Clone()
```

Comment:
That doesn't fully work because now you only locally save the `lastState` when you're receiving, but not when sending, an update. Also, I don't think you need to `Clone` here.

Reply:
So should I fix it to also set it in `sendUpdate` or kick it out until we completely fixed in _go-perun_?

Reply:
I think your fix is good for now, so please also do it in `sendUpdate`.